### ISSUE TITLE
[v1.0.0b] - public recipes not accessible

### ISSUE THREAD

**Issue by econo202** [mealie-recipes/mealie#1880]
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.

### What is the issue you are experiencing?

I'm running Mealie v1.0.0beta-5 with Docker on LAN only.

According to the [documentation](https://hay-kot.github.io/mealie/documentation/recipes/recipes/#recipe-settings), the "public recipe" should be visible without the need to sign in.

What I experience is that  - despite the fact that all the recipes are public - whenever I open Mealie (`http://[local IP:port`]) I get redirected to `http://[local IP:port]/login` and no recipes are shown.

I can access everything only after logging in.

### Deployment

Docker (Linux)

**Comment by hay-kot** [comment]
Can you check the FAQ and see if thay clears things up?

https://nightly.mealie.io/documentation/getting-started/faq/#how-do-private-groups-and-recipes-work

**Comment by andrewdunndev** [comment]
@hay-kot looking at the documentation linked and being a first time user I think that the thing to call out for @econo202 is that you need to go to the recipe, click the three dots, and then click "public link" to get the actual public link to use.

On a fresh deployment of `v1.0.0b` I can validate this does work.

Potentially an enhancement that should be considered is moving the public link from inside that menu item to be a button left of the menu item (sandwiched between edit and menu).

**Comment by SteveDinn** [comment]
This is kind of beyond the scope of this issue, but while reading it, it came to mind:
I'd even suggest changing the concept of a "public link" to a "permanent share". What happens if a public link gets out in the wild and you want to remove it? I see no way to remove or change the URL to the public link. If it were made a share link (even with an infinite expiry date), it could still be removed and re-added with a new URL if that's what you want.

**Comment by hay-kot** [comment]
@SteveDinn, That sort of limits the whole purpose of the public links. They're a building block for public groups so we can provide an interface your external users to view an entire groups recipe catalog. Doing what you described, would make that process _very_ complicated to implement. 

Luckily, the tools to facilitate what you're describing already exist. If a public link was to be exposed in a way that you didn't want, you could always mark that recipe private and use a Private Share Link to generic a unique URL that could then be shared with those your trust. 

I think the biggest thing is that if you're concerns about a public link being exposed, you just shouldn't use a Public Recipe or Group. 

---

@andrewgdunn - I agree we could use some clarity around the public/private links. This issue comes up a lot in the discord. I think the whole menu section could use a refresh with UX in mind.

**Comment by SteveDinn** [comment]
Agreed. I can do what I'm looking to do anyway. How about just having a non-expiring share link option? I'd still be able to remove the link if necessary, but otherwise, it's effectively a public link.

**Comment by econo202** [comment]
Thank you for suggestions. I've checked the [mentioned diagram](https://nightly.mealie.io/documentation/getting-started/faq/#how-do-private-groups-and-recipes-work). I have one group that is not marked as private and all the recipes are marked "public".

- When trying to use a private link everything seems to work, but after generating it, the link isn't copied to clipboard and can't be copied later (but the popup says it was copied).
- When trying to use "public link" from the dropdown menu in the recipe nothing is copied to clipboard. Here also the popup says it was copied.

I've tried both Firefox and Chromium using different profiles (including a new, clean one without any addons). The behavior's the same. Nothing gets copied to clipboard, no matter whether trying to use private or public link.

One thing that I didn't mention previously is that my instance runs only locally and is available under private IP. Maybe that's important in this case.

**Comment by bartmans** [comment]
The clipboard is only accessible with a https site I think. Even locally

**Comment by hay-kot** [comment]
@bartmans is correct. The exact problem is described here

- https://github.com/hay-kot/mealie/issues/1800

TLDR; It's a browser limitation and there's not much I can do about it.

**Comment by econo202** [comment]
Thank you very much! Will have to set up HTTPS then.

**Comment by econo202** [comment]
I was thinking about this issue and came to a partial solution.

The core problem is that nether private nor public link is displayed anywhere. They're designed to be automatically copied to clipboard, which fails on non-HTTPS setups, but it's impossible to see them to copy them manually.

---

The solution would be to show these links somewhere after they're generated.
